cirq-google/cirq_google/engine/validating_sampler.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, List, Optional, Sequence, Union
+
+import cirq
+
+VALIDATOR_TYPE = Callable[[Sequence[cirq.AbstractCircuit], Sequence[cirq.Sweepable]], None]
+
+
+class ValidatingSampler(cirq.Sampler):
+    """Wrapper around cirq.Sampler that performs device validation.
+
+    This sampler will delegate to the wrapping sampler after
+    performing validation on the circuit(s) given to the sampler.
+
+    Args:
+        sampler: sampler wrapped by this object.  After validating,
+            samples will be returned by this enclosed cirq.Sampler.
+        device: cirq.Device that will validate_circuit before sampling.
+        validator: A callable that will do any additional validation
```

Comment:
Should it also take the repetition numbers as well ?

Reply:
Good idea.  Done.